main.c

```diff
@@ -373,6 +380,28 @@ static void get_color(flag_type_t flag_type, float offset, float theta, uint8_t
             gen_color(GFL_BLU, GFL_PNK, theta, GFL_FCT, red, green, blue);
         } 
         break;
+    case FLAG_TYPE_ASEXUAL:
+	if (FLAG_4_0 <= theta && FLAG_4_1 > theta)
+	{ /* black to grey */
+	    theta = 1 - ((theta - FLAG_4_0) / (FLAG_4_1 - FLAG_4_0));
+	    gen_color(ASE_BLA, ASE_GRE, theta, ASE_FCT, red, green, blue);
+	}
+	else if (FLAG_4_1 <= theta && FLAG_4_2 > theta)
+	{ /* grey to white */
+	    theta = 1 - ((theta - FLAG_4_1) / (FLAG_4_2 - FLAG_4_1));
+	    gen_color(ASE_GRE, ASE_WHI, theta, ASE_FCT, red, green, blue);
+	}
+	else if (FLAG_4_2 <= theta && FLAG_4_3 > theta)
+	{ /* white to purple */
+	    theta = 1 - ((theta - FLAG_4_2) / (FLAG_4_3 - FLAG_4_2));
+	    gen_color(ASE_WHI, ASE_PUR, theta, ASE_FCT, red, green, blue);
+	}
+	else if (FLAG_4_3 <= theta && FLAG_4_4 > theta)
+	{ /* purple back to black */
+	    theta = 1 - ((theta - FLAG_4_3) / (FLAG_4_4 - FLAG_4_3));
+	    gen_color(ASE_PUR, ASE_BLA, theta, ASE_FCT, red, green, blue);
+	}
+	break;
```

Comment:
The indentation here is different, also replace tabs with spaces

Reply:
I fixed the indentations and replaced the tabs with spaces